core/src/main/java/io/undertow/server/HttpServerExchange.java

```diff
@@ -480,9 +483,9 @@ public HttpServerExchange setRequestURI(final String requestURI) {
     public HttpServerExchange setRequestURI(final String requestURI, boolean containsHost) {
         this.requestURI = requestURI;
         if (containsHost) {
-            this.state |= FLAG_URI_CONTAINS_HOST;
+            stateUpdater.accumulateAndGet(this, FLAG_URI_CONTAINS_HOST, (state, flag) -> state | flag);
```

Comment:
Perhaps these lambdas could be extracted to static reusable instances -- sometimes lambdas like this in hot paths aren't elided as quickly as one might wish.

Reply:
Isn't this a premature optimization?